osbrain/tests/test_agent_async_requests.py

```diff
@@ -22,114 +23,110 @@ def on_error(agent):
 class Server(Agent):
     def on_init(self):
         self.received = []
+        self.blocked = False
 
 
 class Client(Agent):
     def on_init(self):
         self.received = []
 
 
-@pytest.fixture(scope='function')
-def server_client_late_reply_return():
-    def late_reply(agent, request):
-        agent.received.append(request)
-        time.sleep(1)
-        return 'x' + request
-
+def async_server_client(server_handler=append_received,
+                        client_handler=append_received):
```

Comment:
Also, you always pass as first parameter the `blocked_reply` handler, except for the `yield` test. So I guess that should be the default server handler instead of `append_received`.

Reply:
Yeah, I was trying to make this as generic as possible when I first wrote it, but this makes more sense now :+1: